spec/glueby/internal/wallet/active_record_wallet_adapter_spec.rb

```diff
@@ -322,6 +322,88 @@
     end
   end
 
+  describe '#get_addresses_info' do
+    subject { adapter.get_addresses_info(addresses) }
+
+    before do
+      # Skip callbacks
+      allow(Glueby::Internal::Wallet::AR::Key).to receive(:generate_key)
+      allow(Glueby::Internal::Wallet::AR::Key).to receive(:set_script_pubkey)
+    end
+
+    let!(:keys) do
+      [
+        Glueby::Internal::Wallet::AR::Key.create!({
+          private_key: 'b850e10e8265973f5cd65b385beb1452132dbe10b7eb2a0107c5a04cb2010896',
+          public_key: '0221cc1113cb761d650b27e0e8d4a36802427a53bdd3ed631bd6ffc2fd6b5927b3',
+          script_pubkey: '76a9142f2776cc7fa11adc44a5568604f95283e964a43988ac',
+          label: label,
+          purpose: purpose,
+          wallet_id: wallet.id
+        }),
+        Glueby::Internal::Wallet::AR::Key.create!({
+          private_key: '34d477342334e70ea4259a944e14a3fdd8ce2657c874bcc3a5b94c1a3f754747',
+          public_key: '023b5c432e5d3174900b8c845e8311c1d7fffdd482a60b8c966b13497694bbdb2e',
+          script_pubkey: '76a914a23178289ed6e2a15bd760e7406f85b32d79c17888ac',
+          label: label,
+          purpose: purpose,
+          wallet_id: wallet.id
+        })
+      ]
+    end
+    let(:addresses) { ['15JL32ZJTEeNUT7Fs348errZ8xmavXXhLp', '1FnbjGaaHaitz9FwTpDq4Ss8AkeKpLB5YY'] }
+    let(:purpose) { 'receive' }
+    let(:label) { nil }
```

Comment:
`label` and `purpose` are preferable for different data.

Reply:
What kind of data do you mean?
Would you show me an example?